app/src/main/java/com/example/poste/activities/FolderViewActivity.java

```diff
@@ -72,77 +52,84 @@ protected void onCreate(Bundle savedInstanceState) {
-            if (currentFolder.getPosts().size() > 0) {
-                emptyText.setVisibility(View.GONE);
+        // Create listeners for the folder buttons
+        newBut.setOnClickListener(view -> {
+            Post newPost = PosteApplication.getSelectedFolder().createNewPost(PosteApplication.getSelectedFolder());
+            PosteApplication.setSelectedPost(newPost);
+            Intent intent = new Intent(FolderViewActivity.this, EditPostActivity.class);
+            startActivity(intent);
+        });
+
+        settingsBut.setOnClickListener(view -> {
+            Intent intent = new Intent(FolderViewActivity.this, EditFolderActivity.class);
+            startActivity(intent);
+        });
+
+        // Create listeners for the post buttons
+        openBut.setOnClickListener(view -> {
+            if (PosteApplication.getSelectedPost() != null) {
+                try {
+                    Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(PosteApplication.getSelectedPost().getUrl()));
+                    startActivity(browserIntent);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    Toast.makeText(FolderViewActivity.this, R.string.internal_error, Toast.LENGTH_LONG).show();
+                }
             }
+        });
 
-            // Fill post view (Recycler View)
-            postRecyclerView.setLayoutManager(new LinearLayoutManager(FolderViewActivity.this));
-            postAdapter = new PostAdapter(
-                    new PostAdapter.ClickListener() {
-                        @Override
-                        public void onItemClick(int position, Post post) {
-                            try {
-                                Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(post.getLink()));
-                                startActivity(browserIntent);
-                            } catch (Exception e) {
-                                e.printStackTrace();
-                                Toast.makeText(FolderViewActivity.this, R.string.internal_error, Toast.LENGTH_LONG).show();
-                            }
-                        }
-
-                        @Override
-                        public void onItemLongClick(int position, Post model) {
-//                            Toast.makeText(FolderViewActivity.this, "Share this folder", Toast.LENGTH_LONG).show();
-                        }
-                    },
-                    currentFolder.getPosts()
-            );
-            postRecyclerView.setAdapter(postAdapter);
-            registerForContextMenu(postRecyclerView);
-        } catch (APIException e) {
-            throw new RuntimeException(e);
-        }
-    }
+        editBut.setOnClickListener(view -> {
+            if (PosteApplication.getSelectedPost() != null) {
+                Intent intent = new Intent(FolderViewActivity.this, EditPostActivity.class);
```

Comment:
Does this button edit a post or edit an activity? It seems to call 'EditPost' but we're in Folder view, no?

Reply:
FolderViewActivity is not the view of ALL your folders (that is the DashboardActivity).
FolderViewActivity is the view of what is inside 1 particular folder (i.e. its list of posts).
So this button will open EditPostActivity with the post selected from this screen.
There is no PostViewActivity. FolderView is effectivly the bottom layer of the hierarchy, unless you consider EditPost & EditFolder to be a layer

Reply:
I see. So we're expecting a user to click a post, which only selects it, and then click the edit button? Understood. I did not know that was the intended functionality.